master/buildbot/locks.py

```diff
@@ -133,10 +137,19 @@ def claim(self, owner, access):
 
         assert isinstance(access, LockAccess)
         assert access.mode in ['counting', 'exclusive']
+        assert isinstance(access.count, int)
+        if access.mode == 'exclusive':
+            assert access.count == 1
+        else:
+            assert access.count >= 0
+        if access.count == 0:
```

Comment:
looks like there's a lot of similar patterns in this change

Reply:
ok

Reply:
done